### PR TITLE
Attempt at supporting secrets

### PR DESCRIPTION
**Pull request by abstrctn**
See #16

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -119,6 +120,25 @@ func TestParseAndRunConfig(t *testing.T) {
...
+			env:                  map[string]string{"PLUGIN_ACTION": "deploy", "PLUGIN_TOKEN": validGCPKey, "PLUGIN_SECRETS": `{"var_1":"secretname:latest"}`, "PLUGIN_SERVICE": "my-service", "PLUGIN_IMAGE": "my-image"},
+			cfgExpectedProjectId: "my-project-id",
+			cfgExpectedSecrets:   map[string]string{"var_1": "secretname:latest"},
+			planExpectedFlags:    []string{"--set-secrets", "^:||:^var_1=secretname:latest"},
+			planExpectedOk:       true,
+		},
+		// Test exposing secret as file
+		{
+			cfgExpectedOk:        true,
+			env:                  map[string]string{"PLUGIN_ACTION": "deploy", "PLUGIN_TOKEN": validGCPKey, "PLUGIN_SECRETS": `{"var_1":"/mnt/path:secretname"}`, "PLUGIN_SERVICE": "my-service", "PLUGIN_IMAGE": "my-image"},
```
Comment (oliver006) [inline main_test.go:135]: Is this correct?  --> `"var_1":"/mnt/path:secretname"`

Shouldn't this be `"/mnt/path" : "secretname:latest"` ?

File: main.go [hunk 2]
```diff
@@ -178,6 +185,19 @@ func CreateExecutionPlan(cfg *Config) ([]string, error) {
...
 		}
 
+		if len(cfg.Secrets) > 0 {
+			e := make([]string, 0)
+			for k, v := range cfg.Secrets {
+				e = append(e, fmt.Sprintf(`%s=%s`, k, v))
+			}
+
+			// we're using ":||:" as the separator for the args, let's hope no one puts that in an env variable value
+			sep := ":||:"
```
Comment (oliver006) [inline main.go:195]: nit: you could move this separator out in a const to the top of the file and use it in both places.

**Review by oliver006 — COMMENTED** [review]
I think this makes sense and looks good, just two minor comments.
Could you add something to the README on how to use this? Maybe in a separate block as this is in preview only.
And also add a note that this won't work with Drone 0.8 and/or that Drone 0.8 is no longer supported.

File: main_test.go (hunk 1 above)
Comment (abstrctn) [reply main_test.go]: You're right, thank you!

**Comment by abstrctn** [comment]
Thanks for the feedback @oliver006, and sorry it took me so long to get to it!

I think I've addressed your notes, and added a section to the README about deprecating support for Drone v0.8. (I suppose theoretically it could still work with older versions as long as you don't set any secrets - although it seems unlikely the step would have permission to deploy without any credentials...)

**Comment by oliver006** [comment]
Released as https://github.com/oliver006/drone-cloud-run/releases/tag/v0.4.0

**Pull request merged**